bundles/action/org.openhab.action.mqtt/src/main/java/org/openhab/action/mqtt/internal/MqttMessagePublisher.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.action.mqtt.internal;
+
+import org.openhab.io.transport.mqtt.MqttMessageProducer;
+import org.openhab.io.transport.mqtt.MqttSenderChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implementing Message Producer from MQTT IO service.
+ * 
+ * @author Klaudiusz Staniek
+ * @since 1.8.0
+ */
+public class MqttMessagePublisher implements MqttMessageProducer {
+
+	private static final Logger logger = LoggerFactory.getLogger(MqttMessagePublisher.class);
+
+	private MqttSenderChannel senderChannel;
+	
+	public void publish(String topic, String message) {
+		if (senderChannel == null) {
+			return;
+		}
+		try {
+			senderChannel.publish(topic, message.getBytes());
+		} catch (Exception e) {
+			logger.error("Error publishing message");
```

Comment:
It is actually as comprehensive as the MqttSenderChannel interface description from org.openhab.io.transport.mqtt:
/**
	 * Send a message to the MQTT broker.
	 * 
	 * @param topic
	 *            Topic to publish the message to.
	 * @param message
	 *            message payload.
	 * @throws Exception
	 *             if an error occurs during sending.
	 */
The mqtt transport binding produces a detailed error messages into the log file in case of known issue. This exception handling is more to catch any unexpected situation or exception generated by mqtt pacho library. I can catch specifically the MqttException and log the output of getMessage() method. It it something that would help?
Please refer to http://www.eclipse.org/paho/files/javadoc/org/eclipse/paho/client/mqttv3/MqttException.html